Forwarded: https://github.com/mariadb-corporation/libmarias3/pull/105
From 451aba7ec51bd5d5f227ad1c660046699b20f895 Mon Sep 17 00:00:00 2001
From: Glenn Strauss <gstrauss@gluelogic.com>
Date: Thu, 4 Feb 2021 09:34:26 -0500
Subject: [PATCH] define RTLD_DEFAULT if not defined

---
 storage/maria/libmarias3/src/marias3.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/storage/maria/libmarias3/src/marias3.c
+++ b/storage/maria/libmarias3/src/marias3.c
@@ -33,6 +33,9 @@ ms3_calloc_callback ms3_ccalloc = (ms3_c
 
 /* Thread locking code for OpenSSL < 1.1.0 */
 #include <dlfcn.h>
+#ifndef RTLD_DEFAULT
+#define RTLD_DEFAULT ((void *)0)
+#endif
 static pthread_mutex_t *mutex_buf = NULL;
 #define CRYPTO_LOCK 1
 static void (*openssl_set_id_callback)(unsigned long (*func)(void));
